mavros_extras/src/plugins/mocap_fake_gps.cpp

```diff
@@ -0,0 +1,259 @@
+/**
+ * @brief MocapPoseEstimate plugin
+ * @file mocap_pose_estimate.cpp
+ * @author Tony Baltovski <tony.baltovski@gmail.com>
+ * @author Vladimir Ermakov <vooon341@gmail.com>
+ *
+ * @addtogroup plugin
+ * @{
+ */
+/*
+ * Copyright 2014,2015 Vladimir Ermakov, Tony Baltovski.
+ *
+ * This file is part of the mavros package and subject to the license terms
+ * in the top-level LICENSE file of the mavros repository.
+ * https://github.com/mavlink/mavros/tree/master/LICENSE.md
+ */
+
+#include <mavros/mavros_plugin.h>
+#include <pluginlib/class_list_macros.h>
+#include <eigen_conversions/eigen_msg.h>
+
+#include <geometry_msgs/PoseStamped.h>
+#include <geometry_msgs/TransformStamped.h>
+#include <vicon_bridge/TfDistortInfo.h>
+
+
+namespace mavplugin {
+/**
+ * @brief MocapFakeGPS plugin
+ *
+ * Sends fake GPS from motion capture data to FCU.
+ */
+
+
+class MocapFakeGPSPlugin : public MavRosPlugin
+{
+public:
+	MocapFakeGPSPlugin() :
+		mp_nh("~fake_gps"),
+		uas(nullptr)
+	{ };
+
+	void initialize(UAS &uas_)
+	{
+
+		uas = &uas_;
+
+		mocap_tf_d_sub = mp_nh.subscribe("/tf_distort/out", 1, &MocapFakeGPSPlugin::mocap_tf_d_cb, this);
+
+		mocap_tf_params_sub = mp_nh.subscribe("/tf_distort/info", 1, &MocapFakeGPSPlugin::mocap_tf_params_cb, this);
+
+		mocap_tf_sub = mp_nh.subscribe("/vicon/DJI_450/DJI_450_drop", 1, &MocapFakeGPSPlugin::mocap_tf_cb, this);
+
+		
+	}
+
+	const message_map get_rx_handlers() {
+		return { /* Rx disabled */ };
+	}
+
+private:
+	ros::NodeHandle mp_nh;
+	UAS *uas;
+
+	ros::Subscriber mocap_tf_d_sub;
+	ros::Subscriber mocap_tf_sub;
+	ros::Subscriber mocap_tf_params_sub;
+
+	double old_x;
+	double old_y;
+	double old_z;
+	double old_t;
+	double old_n;
+	double old_e;
+	double gps_eph;
+	double gps_epv;
```

Comment:
I think `Eigen::Vector3d` is more usable here. At least for x,y,z.
And i don't see usage of `old_x` and `old_y`, but instead `old_n`, `old_e`.